### PR TITLE
Forward-ported fixes from python-package.

### LINKED ISSUE #6: Add support for LZMA
We should build a static LZMA lib and then link it to python.

This is a very low prioirty

LZMA is the "native" advance encryption algorithm on macOS

### LINKED ISSUE #27: Enable basic Bash checks.
As with `brink.sh`, enable the following in `pythia`:

```
set -o nounset
set -o errexit
set -o pipefail
```

### LINKED ISSUE #28: Export PIP_INDEX.
From Adi today on #chevah: _one minor update is to also export PIP_INDEX so that you don't have to define it in both paver.conf and pavement.py ... pavement.py can get the URL from the environment_.

### LINKED ISSUE #29: Improved cleanup for distributed packages.
Check if https://github.com/chevah/python-package/pull/158/commits/232fc26c926d4b5c8df6e3c8f8924c5a31cf547a is relevant. More at https://github.com/chevah/python-package/issues/159.

While at it, check for the unusual location of libffi's `include/` sub-dir.

### LINKED ISSUE #31: Rename pythia.sh script.
As discussed on #dev today: _to help with the py3 migration it would help to run both py2 and py3 from the same chevah/server repo. This means that it  would help to have `brink.sh` for python2 and maybe a `pythia.sh` and `pythia.conf` for the python3 environment. I see that  in https://github.com/chevah/pythia pythia.sh is already used for something else... but maybe it can be renamed to `pythia_build.sh` and `pythia_build.sh`_

### PR DESCRIPTION
**Pull request by dumol**
Scope
-----

Bring latest improvements from our `python-package` repo for building Python 2.7 packages.

Changes
-------

Forward-ported updates from python-package:
  * Fixes #28.
  * Fixes #29.
  * Fixes #27.
  * Removed script package management. Required packages are now all installed through GitHub workflows. There is still support for detecting missing packages.

**Drive-by changes**:
  * Renamed `brink.{conf,sh}` as `pythia.{conf.sh}`. Fixes #31.
  * Fixes #6.
  * Updated libffi, zlib, OpenSSL, `pip`, and most Python modules to the newest versions.
  * `cryptography` updated to the latest version not requiring Rust. (Newer versions would practically mean we use binary wheels only for `cryptography`, so at least CentOS 6 would be required.)

I've tried updating Python to latest 3.8 versions. There are 2 main issues:
  * No Windows packages for 3.8.x beyond 3.8.10.
  * Looks like the `readline` patch needs to be updated even for 3.8.10. Thus, I would rather put in the required work when upgrading to a new major version like 3.10.

Testing
-------

Please review changes. Sorry for cramming too much stuff into this PR! Maybe review the commits individually? I've tried to keep things tidy.

Automatic tests should be all green.

### CODE REVIEW

File: src/libffi/chevahbs [hunk 1]
```diff
@@ -12,7 +12,7 @@ chevahbs_getsources() {
     local name=$1
     local ver=$2
     local ext="tar.gz"
-    local link=ftp://sourceware.org/pub/libffi/"$name"-"$ver"."$ext"
+    local link=https://github.com/libffi/libffi/releases/download/v"$ver"/"$name"-"$ver"."$ext"
```
Comment (adiroiban) [inline src/libffi/chevahbs:15]: this is ok for this PR.
but for the long term, we should make sure we have full contol over pythia dependencies.

This mean, keeping copies of things like libffi on bin.chevah.com and have chevahbs script download them from bin.chevah.com

Comment (dumol) [reply src/libffi/chevahbs:15]: Sure, no problem. https://github.com/chevah/pythia/issues/33

File: pythia.conf [hunk 2]
```diff
@@ -1,57 +1,13 @@
...
-    --no-warn-script-location \
-    "
+BASE_REQUIREMENTS='chevah-brink==1.0.6 paver==1.2.4'
+PYTHON_CONFIGURATION='default@3.8.6.3b1a8ba'
+# For production packages there are 2 options:
+BINARY_DIST_URI='https://github.com/chevah/pythia/releases/download'
+#BINARY_DIST_URI='https://bin.chevah.com:20443/production'
+# For testing packages, make sure this one is the last uncommented instance:
+#BINARY_DIST_URI='https://bin.chevah.com:20443/testing'
+PIP_INDEX_URL='https://pypi.chevah.com/simple'
```
Comment (adiroiban) [inline pythia.conf:8]:
```suggestion
PIP_INDEX_URL='https://bin.chevah.com:20443/pypi/simple'
```

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**